framework/include/base/MooseObject.h

```diff
@@ -32,6 +82,8 @@ InputParameters validParams<MooseObject>();
  * Every object that can be built by the factory should be derived from this class.
  */
 class MooseObject :
+  public virtual BlockRestrictableHelper, // see BlockRestrictable.h
+  public virtual BoundaryRestrictableHelper, // see BoundaryRestrictable.h
```

Comment:
I'm against this change.  It doesn't make sense at the most basic level: not all objects even act on geometry.  A list of objects that don't deal with geometry:

1. ScalarKernels
2. GeneralUserObjects
3. TimeIntegrators
4. TimeSteppers
5. Executioners
6. Problem

I'm sure there are others.

MooseObject is EVERYWHERE it must stay light.

In addition: absolutely 100% There should be no virtual inheritance in MOOSE